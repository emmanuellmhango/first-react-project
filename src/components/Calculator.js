import React from 'react';
import Calculate from '../logic/calculate';

class MyCalculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
    this.PutNumber = this.PutNumber.bind(this);
  }

  PutNumber = (event) => {
    this.setState((state) => Calculate(state, event.target.value));
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <div className="calculator-container">
        <div className="calculator-div">
          <div className="display-screen" id="display">
            {total}
            {operation}
            {next}
          </div>
          <button type="button" className="item-ac item-items" value="AC" onClick={this.PutNumber}>AC</button>
          <button type="button" className="item-plus-minus item-items" value="+/-" onClick={this.PutNumber}>+/-</button>
          <button type="button" className="item-modulus item-items" value="%" onClick={this.PutNumber}>%</button>
          <button type="button" className="item-multiply item-controls" value="รท" onClick={this.PutNumber}>รท</button>
          <button type="button" className="item-7 item-items" value="7" onClick={this.PutNumber}>7</button>
          <button type="button" className="item-8 item-items" value="8" onClick={this.PutNumber}>8</button>
          <button type="button" className="item-9 item-items" value="9" onClick={this.PutNumber}>9</button>
          <button type="button" className="item-multiply item-controls" value="x" onClick={this.PutNumber}>x</button>
          <button type="button" className="item-4 item-items" value="4" onClick={this.PutNumber}>4</button>
          <button type="button" className="item-5 item-items" value="5" onClick={this.PutNumber}>5</button>
          <button type="button" className="item-6 item-items" value="6" onClick={this.PutNumber}>6</button>
          <button type="button" className="item-minus item-controls" value="-" onClick={this.PutNumber}>-</button>
          <button type="button" className="item-1 item-items" value="1" onClick={this.PutNumber}>1</button>
          <button type="button" className="item-2 item-items" value="2" onClick={this.PutNumber}>2</button>
          <button type="button" className="item-3 item-items" value="3" onClick={this.PutNumber}>3</button>
          <button type="button" className="item-plus item-controls" value="+" onClick={this.PutNumber}>+</button>
          <button type="button" className="item-0 zero item-items" value="0" onClick={this.PutNumber}>0</button>
          <button type="button" className="item-dot item-items" value="." onClick={this.PutNumber}>.</button>
          <button type="button" className="item-equals item-controls" value="=" onClick={this.PutNumber}>=</button>
        </div>
      </div>
    );
  }
}

export default MyCalculator;
