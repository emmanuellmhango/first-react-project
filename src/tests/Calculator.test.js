import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Snapshot testing', () => {
  test('Testing calculator component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
