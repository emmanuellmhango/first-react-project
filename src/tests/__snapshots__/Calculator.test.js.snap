// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Testing calculator component 1`] = `
<div
  className="calculator-container"
>
  <div
    className="calculator-div"
  >
    <div
      className="display-screen"
      id="display"
    >
      0
       
       
    </div>
    <button
      className="item-ac item-items"
      onClick={[Function]}
      type="button"
      value="AC"
    >
      AC
    </button>
    <button
      className="item-plus-minus item-items"
      onClick={[Function]}
      type="button"
      value="+/-"
    >
      +/-
    </button>
    <button
      className="item-modulus item-items"
      onClick={[Function]}
      type="button"
      value="%"
    >
      %
    </button>
    <button
      className="item-multiply item-controls"
      onClick={[Function]}
      type="button"
      value="รท"
    >
      รท
    </button>
    <button
      className="item-7 item-items"
      onClick={[Function]}
      type="button"
      value="7"
    >
      7
    </button>
    <button
      className="item-8 item-items"
      onClick={[Function]}
      type="button"
      value="8"
    >
      8
    </button>
    <button
      className="item-9 item-items"
      onClick={[Function]}
      type="button"
      value="9"
    >
      9
    </button>
    <button
      className="item-multiply item-controls"
      onClick={[Function]}
      type="button"
      value="x"
    >
      x
    </button>
    <button
      className="item-4 item-items"
      onClick={[Function]}
      type="button"
      value="4"
    >
      4
    </button>
    <button
      className="item-5 item-items"
      onClick={[Function]}
      type="button"
      value="5"
    >
      5
    </button>
    <button
      className="item-6 item-items"
      onClick={[Function]}
      type="button"
      value="6"
    >
      6
    </button>
    <button
      className="item-minus item-controls"
      onClick={[Function]}
      type="button"
      value="-"
    >
      -
    </button>
    <button
      className="item-1 item-items"
      onClick={[Function]}
      type="button"
      value="1"
    >
      1
    </button>
    <button
      className="item-2 item-items"
      onClick={[Function]}
      type="button"
      value="2"
    >
      2
    </button>
    <button
      className="item-3 item-items"
      onClick={[Function]}
      type="button"
      value="3"
    >
      3
    </button>
    <button
      className="item-plus item-controls"
      onClick={[Function]}
      type="button"
      value="+"
    >
      +
    </button>
    <button
      className="item-0 zero item-items"
      onClick={[Function]}
      type="button"
      value="0"
    >
      0
    </button>
    <button
      className="item-dot item-items"
      onClick={[Function]}
      type="button"
      value="."
    >
      .
    </button>
    <button
      className="item-equals item-controls"
      onClick={[Function]}
      type="button"
      value="="
    >
      =
    </button>
  </div>
</div>
`;
